core/src/main/scala/org/apache/spark/deploy/history/HistoryServer.scala

```diff
@@ -69,6 +69,9 @@ class HistoryServer(
 
   private val loaderServlet = new HttpServlet {
     protected override def doGet(req: HttpServletRequest, res: HttpServletResponse): Unit = {
+
+      res.setContentType("text/html;charset=utf-8")
```

Comment:
Thank you, @sarutak . I also confirmed this bug in 3.0.0-rc1.
BTW, 2.4.6-rc1 seems to have no problem. Do you know why we don't have this issue in 2.4.x?

Reply:
I think the reason is that the response header contains `X-Content-Type-Options: nosniff` for `master` and `3.0.0-rc1` while `2.4.x` doesn't.
```
HTTP/1.1 404 Not Found
Date: Wed, 13 May 2020 08:31:43 GMT
Content-Length: 1638
Server: Jetty(9.3.27.v20190418)
```

`X-Content-Type-Options: nosniff` means that the content type should be defined by `Content-Type` header, and it seems to be introduced to `3.0.x` in #23302 for security.

Reply:
Got it. Thank you, @sarutak .